tests/ui/components/search/searchcomponent.js

```diff
@@ -132,15 +135,86 @@ describe('SearchBar component', () => {
     expect(storage.get(StorageKeys.QUERY_TRIGGER)).toEqual(QueryTriggers.SEARCH_BAR);
   });
 
-  it('submitVoiceQuery sets QUERY_TRIGGER to VOICE_SEARCH', () => {
-    const component = COMPONENT_MANAGER.create('SearchBar', {
-      ...defaultConfig,
-      voiceSearch: {
-        enabled: true
-      }
+  describe('Voice search works properly', () => {
+    const events = {};
+    const webkitSpeechRecognitionMock = {
+      addEventListener: jest.fn((eventName, cb) => {
+        events[eventName] = cb;
+      }),
+      start: jest.fn(() => {
+        'start' in events && events.start();
+      }),
+      stop: jest.fn(() => {
+        'stop' in events && events.stop();
+      })
+    };
+
+    let windowSpy;
+    let navigatorSpy;
+
+    beforeEach(() => {
+      windowSpy = jest.spyOn(global, 'window', 'get').mockImplementation(() => ({
+        webkitSpeechRecognition: () => webkitSpeechRecognitionMock,
+        location: {
+          search: {
+            substring: jest.fn()
+          }
+        }
+      }));
+      navigatorSpy = jest.spyOn(global, 'navigator', 'get').mockImplementation(() => ({
+        mediaDevices: {
+          getUserMedia: () => Promise.resolve({
+            getTracks: () => ([{ stop: jest.fn() }])
+          })
+        }
+      }));
+    });
+
+    afterEach(() => {
+      windowSpy.mockRestore();
+      navigatorSpy.mockRestore();
+      Object.values(webkitSpeechRecognitionMock).forEach(value => {
+        (typeof value === 'function') && value._isMockFunction && value.mockClear();
+      });
+    });
+
+    it('submitVoiceQuery sets QUERY_TRIGGER to VOICE_SEARCH', () => {
+      const component = COMPONENT_MANAGER.create('SearchBar', {
+        ...defaultConfig,
+        voiceSearch: {
+          enabled: true
+        }
+      });
+      mount(component);
+      component.submitVoiceQuery();
+      expect(storage.get(StorageKeys.QUERY_TRIGGER)).toEqual(QueryTriggers.VOICE_SEARCH);
+    });
+
+    it('Clicking the voice search button starts voice search', async () => {
+      const component = COMPONENT_MANAGER.create('SearchBar', {
+        ...defaultConfig,
+        voiceSearch: {
+          enabled: true
+        }
+      });
+      mount(component);
+      await component._voiceSearchController.handleIconClick();
+      expect(webkitSpeechRecognitionMock.start).toHaveBeenCalledTimes(1);
+      expect(webkitSpeechRecognitionMock.stop).toHaveBeenCalledTimes(0);
+    });
+
+    it('Clicking the voice search button twice starts then stops voice search', async () => {
+      const component = COMPONENT_MANAGER.create('SearchBar', {
+        ...defaultConfig,
+        voiceSearch: {
+          enabled: true
+        }
+      });
+      mount(component);
+      await component._voiceSearchController.handleIconClick();
+      await component._voiceSearchController.handleIconClick();
+      expect(webkitSpeechRecognitionMock.start).toHaveBeenCalledTimes(1);
```

Comment:
maybe we can switch lines 215 and 216?

Reply:
Sure, I just combined these into one test